apps/designer/app/shared/copy-paste/copy-paste-instance.ts

```diff
@@ -0,0 +1,69 @@
+import ObjectId from "bson-objectid";
+import { useEffect } from "react";
+import { z } from "zod";
+import { Instance, Props, Styles } from "@webstudio-is/project-build";
+import { utils } from "@webstudio-is/project";
+import { startCopyPaste } from "./copy-paste";
+
+const InstanceCopyData = z.object({
+  instance: Instance,
+  props: Props,
+  styles: Styles,
+});
+
+export type InstanceCopyData = z.infer<typeof InstanceCopyData>;
+
+type InstanceCopyPasteProps = {
+  selectedInstanceData?: InstanceCopyData | undefined;
+  allowAnyTarget?: boolean;
+  onPaste: (data: InstanceCopyData) => void;
+  onCut: (instance: Instance) => void;
+};
+
+// to make it easier to remove React from canvas if we need to
+export const useInstanceCopyPaste = (props: InstanceCopyPasteProps): void => {
+  const { selectedInstanceData, allowAnyTarget, onPaste, onCut } = props;
+  useEffect(() => {
+    selectedInstanceData;
+    onPaste;
+    allowAnyTarget;
```

Comment:
wow, what is this doing?

Reply:
Looks like paste works for non inputs with it

Reply:
Ah, gotcha. Forgot to remove